src/Microsoft.PowerShell.Commands.Utility/commands/utility/MatchString.cs

```diff
@@ -1176,14 +1176,21 @@ public new int[] Context
             }
         }
 
+        /// <summary>
+        /// Gets or sets a value indicating if only string values containing just the matched text should be returned.
+        /// If not (default) return MatchInfo (or bool objects, when Quiet is passed).
+        /// </summary>
+        [Parameter]
+        public SwitchParameter OnlyMatching { get; set; }
```

Comment:
```suggestion
        [Alias('o')]
        public SwitchParameter OnlyMatching { get; set; }
```

We may want to add the explicit alias here as mentioned in the related issue.

Reply:
Added the alias.

Reply:
I think the alias should be `om` which is much less ambiguous (`-o` matches every param that starts with `o`) and is consistent with `-ob` for `-OutBuffer` and `-ov` for `OutVariable`.

Reply:
@rkeithhill I'd agree, but I think the intent in the original issue was to match `grep`'s existing `-o` parameter.

Reply:
Considering all the parameters on Select-String that do not match grep, is it worth it to try to force-fit just this one?

Reply:
I agree that `-om` is more in keeping with the PowerShell style. Does @PowerShellTeam have a preference for what it should be?

Reply:
I too agree that `-om` is better; to get a definitive ruling the committee would need to be pinged, but we mere mortals can't do that, I think; @SteveL-MSFT, any thoughts?

@vexx32, re returning the text from all match instances on the line with `-AllMatches`: yes, that was the intent; it's how `grep -o` does it too (albeit invariably); e.g., `'foo' | grep -o 'o'`